src/analysis/seminorm.lean

```diff
@@ -312,6 +316,59 @@ nonneg_of_mul_nonneg_left h zero_lt_two
 
 lemma sub_rev : p (x - y) = p (y - x) := by rw [←neg_sub, p.neg]
 
+variables {α : Type*} [semilattice_sup α] {a b c d : α}
+variables (h₁ : a = b)
+
+lemma mul_sup {a b c : ℝ} (h₁ : 0 ≤ a) : a * (b ⊔ c) = (a * b) ⊔ (a * c) :=
+begin
+  cases le_total b c with h h,
+  { simp [sup_eq_max, max_eq_right h, max_eq_right (mul_le_mul_of_nonneg_left h h₁)] },
+  { simp [sup_eq_max, max_eq_left h, max_eq_left (mul_le_mul_of_nonneg_left h h₁)] },
+end
+
+instance : has_sup (seminorm 𝕜 E) :=
+{ sup := λ p q,
+  { to_fun := p ⊔ q,
+    triangle' :=
+      begin
+        intros x y,
+        simp,
+        split,
+        { apply le_trans (p.triangle x y),
+          exact add_le_add (le_sup_left) (le_sup_left) },
+        apply le_trans (q.triangle x y),
+        exact add_le_add le_sup_right le_sup_right,
+      end,
+    smul' :=
+      begin
+        intros x v,
+        simp,
+        rw mul_sup (norm_nonneg x),
+        rw [p.smul x v, q.smul x v],
+      end }}
```

Comment:
This one golfs too:
```suggestion
    smul' := λ x v, (congr_arg2 (⊔) (p.smul x v) (q.smul x v)).trans $ mul_sup (norm_nonneg x) }
```
Note that `congr_arg2 (⊔)` is precisely the `sup_eq_sup` lemma you had before!